src/overlays/actors/ovl_En_Gg/z_en_gg.c

```diff
@@ -74,62 +95,719 @@ static DamageTable D_80B36C38 = {
+        case 0xCE5:
+            return 0xCE6;
+        case 0xCE6:
+            return 0xCE8;
+        case 0xCE8:
+            return 0xCE9;
+        case 0xCE9:
+            return 0xCEC;
+        case 0xCEC:
+            return 0xCED;
+    }
+    return 0;
+}
+
+void func_80B3584C(EnGg* this) {
+    this->unk_30A += 0x72C;
+    this->actor.shape.yOffset = Math_SinS(this->unk_30A) * 100.0f;
+    this->actor.shape.shadowScale = 30.0f - (Math_SinS(this->unk_30A) * 5.0f);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+void func_80B358D8(EnGg* this, GlobalContext* globalCtx) {
+    s16 sp1E = this->skelAnime.curFrame;
+    s16 frame = Animation_GetLastFrame(sAnimations[this->unk_2DA].animation);
+
+    if ((this->unk_2E6 == 14) && (sp1E == frame)) {
+        this->unk_2E6 = 15;
+        Actor_ChangeAnimation(&this->skelAnime, sAnimations, 15);
+    }
+}
+
+void func_80B35968(EnGg* this, GlobalContext* globalCtx) {
+    if (this->unk_344.unk_34 != NULL) {
+        this->unk_344.unk_34(&this->unk_344, globalCtx);
+    } else {
+        if ((this->unk_2DA == 11) || (this->unk_2DA == 14)) {
+            this->unk_344.unk_30 = this->unk_2DA;
+        }
+        func_80B363E8(&this->unk_344, globalCtx, &this->unk_320, &this->unk_32C, &this->unk_338);
+    }
+}
+
+void func_80B359DC(EnGg* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (this->actor.xzDistToPlayer < 200.0f) {
+        if (this->unk_306 == 0) {
+            if (player->stateFlags2 & 0x8000000) {
+                this->unk_306 = 1;
+                play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            }
+        } else if (!(player->stateFlags2 & 0x8000000)) {
+            this->unk_306 = 0;
+        }
+
+        if ((player->transformation == PLAYER_FORM_HUMAN) && (globalCtx->msgCtx.unk1202A == 3) &&
+            (globalCtx->msgCtx.unk1202E == 7)) {
+            if (!(gSaveContext.weekEventReg[19] & 0x80)) {
+                gSaveContext.weekEventReg[19] |= 0x80;
+            }
+            this->unk_307 = 1;
+        }
+
+        if (ActorCutscene_GetCanPlayNext(this->unk_2DC)) {
+            ActorCutscene_Start(this->unk_2DC, &this->actor);
+            this->unk_307 = 0;
+        } else {
+            if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+                ActorCutscene_Stop(0x7C);
+            }
+
+            if (this->unk_307 != 0) {
+                ActorCutscene_SetIntentToPlay(this->unk_2DC);
+            }
+        }
+    } else {
+        this->unk_307 = 0;
+    }
+}
+
+void func_80B35B24(EnGgStruct* ptr, GlobalContext* globalCtx) {
+    ptr->unk_34 = NULL;
+    ptr->unk_38 = NULL;
+    ptr->unk_3C = 0;
+    ptr->unk_40 = 0;
+    ptr->unk_44 = 0;
+}
+
+void func_80B35B44(EnGgStruct* ptr, GlobalContext* globalCtx) {
+    if (ptr->unk_30 == 0xE) {
+        ptr->unk_40++;
+        if (ptr->unk_40 >= 0x47) {
```

Comment:
```suggestion
        if (ptr->unk_40 > 0x46) {
```
You can replace other instances of `0x47` like this in this function, since `0x46` seems to be the important number